libraries/AP_HAL_PX4/RCInput.cpp

```diff
@@ -118,6 +118,10 @@ void PX4RCInput::_timer_tick(void)
     if (orb_check(_rc_sub, &rc_updated) == 0 && rc_updated) {
         pthread_mutex_lock(&rcin_mutex);
         orb_copy(ORB_ID(input_rc), _rc_sub, &_rcin);
+        if (_rcin.rssi != 0 || _rssi != -1) {
```

Comment:
no, it is supposed to be an ||
the reason is that there is no flag from the backend on whether the RSSI field is valid. So we use a single non-zero value to mean "yes, we have RSSI input"